code/binary-tree/src/test/java/ru/unn/agile/binaryTree/model/BinaryTreeFunctionalTests.java

```diff
@@ -0,0 +1,109 @@
+package ru.unn.agile.binarytree.model;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Random;
+
+import static org.junit.Assert.*;
+
+public class BinaryTreeFunctionalTests {
+    private BinaryTree binaryTree;
+
+    private int     simpleKey   = 1;
+    private String  simpleData  = "FirstSimpleText";
+
+    private int bound = 1000;
+    private int arrSize = 10;
+    private static Random randomGen = new Random();
+
+    @Before
+    public void setUp() {
+        binaryTree = new BinaryTree();
+    }
+
+    @Test
+    public void canAddAfterDeletingRoot() {
+        binaryTree.add(simpleKey, simpleData);
+        final int    secondKey  = 2;
+        final String secondData = "SecondSimpleText";
+
+        assertTrue(binaryTree.remove(simpleKey));
+
+        binaryTree.add(secondKey, secondData);
```

Comment:
You don't have an assertion after the act you do in this test. That's incorrect.
It's better to remove assertion about removal of simpleKey (because it's useless from the test perspective) and add an assertion that you can find secondData by secondKey after add